### PR TITLE
Update Cargo.lock so that it points to new version of branch aleph-v0.23

### PR DESCRIPTION
**Pull request by mike1729**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- I have added tests
- I have made neccessary updates to the Infrastructure
- I have made corresponding changes to the existing documentation
- I have created new documentation
- I have bumped `spec_version` and `transaction_version`
- I have bumped aleph-client version if relevant

### CODE REVIEW

**Comment by pmikolajczyk41** [comment]
you probably have to set `syn = "=1.0.86"`, since the newest version have some problems incompatible with Substrate

https://bytemeta.vip/repo/paritytech/substrate/issues/11706
https://substrate.stackexchange.com/questions/3479/error-building-substrate-contract-nodeerrore0603-module-group-is-private

**Review by pmikolajczyk41 — COMMENTED** [review]
Don't we want to bump also Cargo.locks in other packages (outside main workspace)? Like aleph-client/cliain/e2e-tests etc.

**Comment by mike1729** [comment]
> Don't we want to bump also Cargo.locks in other packages (outside main workspace)? Like aleph-client/cliain/e2e-tests etc.

The change in pallet contracts is internal so there is no point, but `syn` may be the reason?

File: bin/runtime/src/lib.rs [hunk 1]
```diff
@@ -112,7 +112,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     spec_name: create_runtime_str!("aleph-node"),
     impl_name: create_runtime_str!("aleph-node"),
     authoring_version: 1,
-    spec_version: 21,
+    spec_version: 22,
```
Comment (pmikolajczyk41) [inline bin/runtime/src/lib.rs:115]: this one will be probably have to be bumped to `25`

**Review by pmikolajczyk41 — APPROVED** [review]
as long as it compiles and passes tests, I'm good

**Pull request merged**